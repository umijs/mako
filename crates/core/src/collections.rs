pub use rustchash::{FxHashMap as HashMap, FxHashSet as HashSet};

mod rustchash {
    use std::collections::{HashMap, HashSet};
    use std::hash::BuildHasherDefault;

    use rustc_hash::FxHasher;

    pub type FxRandomState = BuildHasherDefault<FxHasher>;

    pub type FxHashMap<K, V> = HashMap<K, V, FxRandomState>;

    pub type FxHashSet<V> = HashSet<V, FxRandomState>;
}

#[macro_export(local_inner_macros)]
macro_rules! hashmap {
    (@single $($x:tt)*) => (());
    (@count $($rest:expr),*) => (<[()]>::len(&[$(hashmap!(@single $rest)),*]));

    ($($key:expr => $value:expr,)+) => { hashmap!($($key => $value),+) };
    ($($key:expr => $value:expr),*) => {
        {
            let _cap = hashmap!(@count $($key),*);
            let mut _map = $crate::collections::HashMap::default();
            $(
                let _ = _map.insert($key, $value);
            )*
            _map
        }
    };
}
