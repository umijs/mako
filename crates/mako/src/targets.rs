use std::collections::HashMap;

use mako_core::swc_ecma_preset_env::Targets as SwcPresetEnvTargets;

pub fn swc_preset_env_targets_from_map(map: HashMap<String, f32>) -> SwcPresetEnvTargets {
    let serialized_str = serde_json::to_string(&map).unwrap();
    let targets: SwcPresetEnvTargets = serde_json::from_str(&serialized_str).unwrap();
    targets
}

#[cfg(test)]
mod tests {
    use std::collections::HashMap;

    use super::swc_preset_env_targets_from_map;
    use crate::assert_debug_snapshot;

    #[test]
    fn test_swc_preset_env_targets() {
        let map = HashMap::from([("chrome".to_string(), 80.0)]);
        let targets = swc_preset_env_targets_from_map(map);
        assert_debug_snapshot!(&targets);
    }
}
