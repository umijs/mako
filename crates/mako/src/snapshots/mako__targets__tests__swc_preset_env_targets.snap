---
source: crates/mako/src/targets.rs
assertion_line: 43
expression: "&targets"
---
Versions(
    BrowserData {
        chrome: Some(
            Version {
                major: 80,
                minor: 0,
                patch: 0,
            },
        ),
        chrome_android: None,
        firerfox_android: None,
        opera_android: None,
        quest: None,
        react_native: None,
        and_chr: None,
        and_ff: None,
        op_mob: None,
        ie: None,
        edge: None,
        firefox: None,
        safari: None,
        node: None,
        ios: None,
        samsung: None,
        opera: None,
        android: None,
        electron: None,
        phantom: None,
        opera_mobile: None,
        rhino: None,
        deno: None,
        hermes: None,
        oculus: None,
        bun: None,
    },
)
